app/src/main/java/org/wikipedia/imagesearch/Encoder.java

```diff
@@ -0,0 +1,73 @@
+package org.wikipedia.imagesearch;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.util.Base64;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Created by mnhn3 on 2018-04-12.
+ */
+
+public class Encoder {
+
+    public String encodeFileToBase64Binary(File file) {
+        String encodedFile = null;
+        try {
+            FileInputStream fileInputStreamReader = new FileInputStream(file);
+            byte[] bytes = new byte[(int)file.length()];
+            fileInputStreamReader.read(bytes);
+            //encodedFile = new String(Base64.encodeBase64(bytes), "UTF-8");
+            encodedFile = Base64.encodeToString(bytes, Base64.NO_WRAP);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return encodedFile;
+    }
+
+    //Encodes a file given a content Uri as input
+    public String encodeUriToBase64Binary(Context context, Uri uri) {
```

Comment:
Could you elaborate on the purpose of this method? Why do we need to encode?

Reply:
Currently the images are stored as a URI object, but the API request requires a string that is base64 encoded, so this method provides that conversion